### PR TITLE
Enable autofix for config.get/require to meta_get/meta_require migration

### PR DESCRIPTION
**Pull request by venkaa28**
Dont merge yet, depends on this pr: https://github.com/dbt-labs/fs/pull/7007

This PR enables and improves the autofix rule that migrates custom config access from deprecated patterns to the
  new meta-specific methods.

  What changed:

  - Enabled the previously commented-out move_custom_config_access_to_meta_sql autofix
  - Created improved implementation that correctly handles both config.get() and config.require()
  - Added comprehensive test coverage for edge cases

  Key improvements:

  - Correct syntax: Replaces config.get('custom_key') → config.meta_get('custom_key') (not the incorrect
  .get('meta').key)
  - Support for require: Also handles config.require('custom_key') → config.meta_require('custom_key')
  - Preserves arguments: Maintains defaults and validators in refactored code
  - Smart detection: Only refactors custom configs, leaves dbt-native configs unchanged
  - Better warnings: Detects variable shadowing and chained access patterns that need manual review

  Example transformation:

  -- Before
  {% set team = config.get('team_config') %}
  {% set env = config.get('environment', 'dev') %}
  {% set required = config.require('custom_key') %}

  -- After (autofix)
  {% set team = config.meta_get('team_config') %}
  {% set env = config.meta_get('environment', 'dev') %}
  {% set required = config.meta_require('custom_key') %}

  This autofix helps users migrate away from the deprecated config access pattern that will emit warnings in Fusion.

### CODE REVIEW

File: tests/integration_tests/dbt_projects/project1_expected/models/example/model_with_custom_configs.sql [hunk 1]
```diff
@@ -3,14 +3,14 @@
     meta={'existing_meta_config': 'existing_meta_config', 'custom_config': 'custom_config', 'custom_config_int': 2, 'custom_config_list': ['a', 'b', 'c'], 'custom_config_dict': {'a': 1, 'b': 2, 'c': 3}}
 ) }}
 
-{{ config.get('custom_config') }}
+{{ config.meta_get('custom_config') }}
```
Comment (chayac) [inline tests/integration_tests/dbt_projects/project1_expected/models/example/model_with_custom_configs.sql]: Merging this is blocked until `config.meta_get` is implemented in Fusion, right?

File: tests/integration_tests/dbt_projects/project1_expected/models/example/model_with_custom_configs.sql [hunk 2]
```diff
@@ -3,14 +3,14 @@
...
+{{ config.meta_get('custom_config', 'default_value') }}
 
 {{ config.get('materialized') }}
 
 {{ config.get('materialized', 'default_value') }}
 
-{{ config.get('custom_config_dict').a }}
+{{ config.meta_get('custom_config_dict').a }}
 
-{{ config.get('custom_config_dict').get('a', 'default_value') }}
```
Comment (MichelleArk) [inline tests/integration_tests/dbt_projects/project1_expected/models/example/model_with_custom_configs.sql]: Tested this out with a few more edge cases around specifying the 'default' kwarg explicitly:

expected inputs: 
```
{{ config.get('custom_config', default='default_value') }}

{{ config.get('custom_config', default=var.get('my_var')) }}

{{ config.get('custom_config', default=dest_columns | map(attribute=\"quoted\") | list) }}
```

and outputs:
```
{{ config.meta_get('custom_config', default='default_value') }}

{{ config.meta_get('custom_config', default=var.get('my_var')) }}

{{ config.meta_get('custom_config', default=dest_columns | map(attribute=\"quoted\") | list) }}
```

all worked as expected!

would suggest adding these cases as tests though

File: tests/unit_tests/test_config_meta_autofix.py
[the remark this answers is not in the document]
Comment (ChenyuLInx) [reply tests/unit_tests/test_config_meta_autofix.py]: That works! feels like a small enough case
